server/ec2spotmanager/tasks.py

```diff
@@ -234,389 +230,438 @@ def _start_pool_instances(pool, config, count=1):
 
         userdata = LaniakeaCommandLine.handle_tags(userdata, ec2_userdata_macros)
         if not userdata:
-            logger.error("[Pool %d] Failed to compile userdata.", pool.id)
 
+            logger.error("[Pool %d] Failed to compile userdata.", pool.id)
             entry = PoolStatusEntry()
-            entry.type = POOL_STATUS_ENTRY_TYPE['config-error']
+            entry.type = EC2_POOL_STATUS_ENTRY_TYPE['config-error']
             entry.pool = pool
             entry.isCritical = True
-            entry.msg = "Configuration error: Failed to compile userdata"
+            entry.msg = "Configuration error: %s" % traceback.format_exc()
```

Comment:
Do not expose a traceback here. They disclose server paths and are unsuitable for production error messages. Instead, you can output the backtrace if the Django instance is running in DEBUG mode, or output it to one of the configured log files.